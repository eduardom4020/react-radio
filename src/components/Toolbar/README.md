# Toolbar
>This component is a view renders the application's toolbar, that is the top bar.

### Props
These are the props accepted by the component, pay attention to the description, inputs and behaviour:

---

##### height *(string | number)*
*The height of the toolbar. By default it's four times status bar height.*

**example**: 
```
<Toolbar height={100} /> 
```

---

##### color *(string)*
*The background color in wich the footer will be filled. You should pass to it any color code accepted by CSS, like hexadecial, rgb or the color name. By default it's grey*

**example**: 
```
<Footer color='yellow' />
```

---