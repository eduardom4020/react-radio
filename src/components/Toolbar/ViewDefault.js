import styled from 'styled-components';
import { View, StatusBar } from 'react-native';

const ToolbarViewDefault = styled(View)`
    width: 100%;
    height: ${props => props.height == null ? '15%' : typeof(props.height) === 'number' ? `${props.height}px` : props.height };
    background-color: ${props => props.color || 'grey'};
    flex-direction: row;
    justify-content: space-around;
    align-items: center;
    padding: ${( StatusBar.currentHeight || 0 )}px 0 0 0;
`;

export default ToolbarViewDefault;