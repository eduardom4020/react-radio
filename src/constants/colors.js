export const BLACK_LIKE = '#20202b';
export const DARK_GREY = '#2e2e3a';
export const MEDIUM_GREY = '#a3acbe';
export const MEDIUM_ORANGE = '#eeae61';
export const METALIC_BLACK = '#191313';