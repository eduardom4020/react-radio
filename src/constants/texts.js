export const APP_TITLE = 'STATIONS';