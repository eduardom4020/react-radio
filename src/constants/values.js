export const DEFAULT_IMAGE_BT_SIZE = 192;
export const DEFAULT_ICON_BT_SIZE = 24;
export const DEFAULT_RADIO_PLAYING = 'CURRENTLY PLAYING'