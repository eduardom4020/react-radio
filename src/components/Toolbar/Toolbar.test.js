import React from 'react';
import { Text, View } from 'react-native';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { DARK_GREY } from '~/src/constants/colors';
import Toolbar from '~/src/components/Toolbar';

describe('Toolbar', () => {
    test('Render?', () => {
        const component = shallow(
            <Toolbar />
        );

        expect(toJson(
            component, {
                noKey: false,
                mode: 'deep'
            }
        )).toMatchSnapshot();
    });

    test('Render with custom color?', () => {
        const component = shallow(
            <Toolbar color={DARK_GREY} />
        );

        expect(toJson(
            component, {
                noKey: false,
                mode: 'deep'
            }
        )).toMatchSnapshot();
    });

    test('Render with custom height?', () => {
        const component = shallow(
            <Toolbar height={200} />
        );

        expect(toJson(
            component, {
                noKey: false,
                mode: 'deep'
            }
        )).toMatchSnapshot();
    });
    
    test('Render with children?', () => {
        const component = shallow(
            <Toolbar color={DARK_GREY} >
                <View><Text>Test1</Text></View>
                <Text>Test2</Text>
            </Toolbar>
        );

        expect(toJson(
            component, {
                noKey: false,
                mode: 'deep'
            }
        )).toMatchSnapshot();
    });
});