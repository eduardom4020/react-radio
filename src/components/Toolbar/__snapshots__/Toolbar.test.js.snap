// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar Render with children? 1`] = `
<StyledComponent
  color="#2e2e3a"
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
    width: 100%;
    height: ",
          [Function],
          "px;
    background-color: ",
          [Function],
          ";
    flex-direction: row;
    justify-content: space-around;
    align-items: center;
    padding: ",
          "42",
          "px 0 0 0;
",
        ],
      },
      "displayName": "Styled(View)",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": [Function],
      "toString": [Function],
      "usesTheme": false,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  <View>
    <Text>
      Test1
    </Text>
  </View>
  <Text>
    Test2
  </Text>
</StyledComponent>
`;

exports[`Toolbar Render with custom color? 1`] = `
<StyledComponent
  color="#2e2e3a"
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
    width: 100%;
    height: ",
          [Function],
          "px;
    background-color: ",
          [Function],
          ";
    flex-direction: row;
    justify-content: space-around;
    align-items: center;
    padding: ",
          "42",
          "px 0 0 0;
",
        ],
      },
      "displayName": "Styled(View)",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": [Function],
      "toString": [Function],
      "usesTheme": false,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Toolbar Render with custom height? 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
    width: 100%;
    height: ",
          [Function],
          "px;
    background-color: ",
          [Function],
          ";
    flex-direction: row;
    justify-content: space-around;
    align-items: center;
    padding: ",
          "42",
          "px 0 0 0;
",
        ],
      },
      "displayName": "Styled(View)",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": [Function],
      "toString": [Function],
      "usesTheme": false,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  height={200}
/>
`;

exports[`Toolbar Render? 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
    width: 100%;
    height: ",
          [Function],
          "px;
    background-color: ",
          [Function],
          ";
    flex-direction: row;
    justify-content: space-around;
    align-items: center;
    padding: ",
          "42",
          "px 0 0 0;
",
        ],
      },
      "displayName": "Styled(View)",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": [Function],
      "toString": [Function],
      "usesTheme": false,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
